src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List getAll(Class type) {
+        if (type == Bulldozer.class) {
+            return new BulldozerProducer().get();
```

Comment:
You can avoid casting this way
```suggestion
            return new ArrayList<>(new BulldozerProducer().get());
```